src/main/scala/cromwell/binding/WdlBinding.scala

```diff
@@ -115,6 +114,11 @@ object WdlBinding {
     }
   }
 
+  /* All MemberAccess ASTs that are not contained in other MemberAccess ASTs */
+  def findTopLevelMemberAccesses(expr: AstNode) = WdlBinding.findAstsWithTrail(expr, "MemberAccess").filter {
+    case (k, v) => !(v map {case a:Ast => a.getName} contains "MemberAccess")
```

Comment:
Another way of phrasing what I'm getting at is that function should be doing the heavy lifting (ideally in a referentially transparent way) and not closures. Combinators being applied to a pipe of operations should ideally be pretty simple and call out to functions rapidly. This way the line that's actually triggering the Real Work is just something like:

     x map { foo } filter {_ > 5} flatMap {something} ...

And you can pack a lot of operations in a single easy to read line. As long as things like `foo` and `something` are descriptively named there's not often a need to even look and see how they're implemented.

I hear you on the notion that now you'd just have lots of little functions everywhere but this is an area where I think OO/FP hybrids really start to show their weak spots as people combine their dislike of lots of functions with their enjoyment of higher order functions.